csharp/Array_Lists_Dictionaries/Readme.md

```diff
@@ -0,0 +1,133 @@
+
+Eine Variable von Typ ```List``` legt man so an:
+
+```csharp
+List<int> alterMeinerFreunde = new List<int>();
+```
+
+Hier haben wir eine ```List``` von Typ ```int``` angelegt, also eine Liste wo Ganzzahlwerten gespeichert werden können. Was wir _nicht_ angegeben haben wir groß die Liste ist.
+
+So fügen wir paar Werte dazu:
+
+```csharp
+alterMeinerFreunde.Add(10);
+alterMeinerFreunde.Add(12);
+alterMeinerFreunde.Add(11);
+alterMeinerFreunde.Add(9);
+alterMeinerFreunde.Add(10);
+```
+
+Siehst du? Du kannst mit der Funktion ```Add``` Werte hinzufügen. Aber nicht nur das, du kannst die Werte auch wieder entfernen:
+
+```csharp
+alterMeinerFreunde.Remove(9);
+```
+
+Es gibt viele Funktionen du auf einer List ausführen kannst, hier paar Beispiele:
+
+```csharp
+int anzahl = alterMeinerFreunde.Count; // Gibt zurück wie viel Einträge sind in die Liste
+alterMeinerFreunde.Sort(); // Ordnet die Liste an
+alterMeinerFreunde.Clear(); // Löscht alle Einträge aus die Liste
+bool gibtEsEinAlterZehn = alterMeinerFreunde.Contains(10); // Gibt zurück in die Variable gibtEsEinAlterZehn ob ein Eintrag 10 existiert
+```
+
+Wenn du einen Wert auslesen möchtest, kannst das so machen wie bei Array:
+
+```csharp 
+int meinesFreundAlter = alterMeinerFreunde[0];
+```
+
+Es gibt auch andere und bessere Varianten um Werte auszulesen. Dazu aber ein anderes mal mehr.
+
+### Aufgabe
+
+Schreibe in einer Variable Namen von drei deiner besten Freunde oder Freundinnen. Sortiere diese von A bis Z. Gib jede dieser Namen in einer _Console_ aus, in dem du die vorher angelegte Variable verwendest.
+
+
+## Dictionaries
+
+Du hast gesehen, dass wir in vorherigen Beispielen zwar das Alter aber nicht die Namen dazu gespeichert haben. Dazu geeignet sich ```Dictionary```. Dictionary ist eine spezielle Liste, mit welcher man an einer Position zwei Werte speichern kann, eine Wert als _Key_ und einen weiteren als _Value_.
+
+```csharp
+Dictionary<string,int> altersBuch = new Dictionary<string,int>();
+altersBuch.Add("Toni", 39);
+altersBuch.Add("Daniel", 38);
+```
+
+Wichtig ist hier zu sagen dass das _key_ eindeutig sein muss. Das hier wird hier nicht funktionieren:
+
+```csharp
+Dictionary<string,int> altersBuch = new Dictionary<string,int>();
+altersBuch.Add("Toni", 39);
+altersBuch.Add("Daniel", 38);
+altersBuch.Add("Toni", 25); //hier wird bei ausführen des Programm eine Fehler passieren, weil schon Toni hinzugefügt haben.
+```
+
+Wenn du einen Wert aus Dictionary auslesen möchtest kannst du es folglich machen:
+```csharp
+var alterVonDaniel = altersBuch["Daniel"];
+```
+
+### Aufgabe
+
+Schreibe in einer Variable Vorname, Nachname und das Alter von drei besten Freunden oder Freundinnen. Gebe das Alter von einen diesen Namen in einer _Console_ aus in dem du die vorher angelegte Variable verwendest.

```

Comment:
Isn't knowledge on classes necessary to solve this, which aren't introduced yet?